src/main/java/uk/ac/ebi/ega/accession/configuration/security/CustomUserAuthenticationConverter.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package uk.ac.ebi.ega.accession.configuration.security;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.provider.token.UserAuthenticationConverter;
+import uk.ac.ebi.ega.accession.user.AccessioningUser;
+import uk.ac.ebi.ega.accession.user.AccessioningUserRepository;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * This class provides custom authorities for the EGA AAI authenticated user.
+ */
+public class CustomUserAuthenticationConverter implements UserAuthenticationConverter {
```

Comment:
After giving some thought, this is not the proper way of handle this. The converter is just to pass the map to the authorization token it is not the proper way of getting this information (altough you can do it as you've done). The proper way would be to have configured correctly this: https://docs.spring.io/spring-security-oauth2-boot/docs/current/api/org/springframework/boot/autoconfigure/security/oauth2/resource/AuthoritiesExtractor.html